src/models/calendar.ts

```diff
@@ -1,15 +1,37 @@
 import RestfulModel, { SaveCallback } from './restful-model';
 import { GetCallback } from './restful-model-collection';
 import Attributes from './attributes';
+import NylasConnection from '../nylas-connection';
 
-export default class Calendar extends RestfulModel {
-  name?: string;
-  description?: string;
+export interface CalendarProperties {
+  name: string;
+  description: string;
+  location: string;
+  timezone: string;
   readOnly?: boolean;
-  location?: string;
-  timezone?: string;
   isPrimary?: boolean;
   jobStatusId?: string;
+}
+
+export default class Calendar extends RestfulModel
+  implements CalendarProperties {
+  name!: string;
```

Comment:
I added bang (!) as a modifier because we guarantee to set these values somehow -- either via the props that are sent in (which contain the strict properties) or if no props are sent in via setting default values.